### PR TITLE
feat: add modal to send email

### PR DESCRIPTION
**Pull request by andrewlew1s**
As part of the work effort outlined here: https://github.com/holistic-web/portfolio-fabricator/issues/11 we need to add a modal that can be shown in the outputted site.

It will need to submit this in some way to send an email and we will need to allow the user to configure whether or not this feature is enabled.

A basic modal has been added to the site. This is what we need to do next:

### UI Improvements 

- [x] Message form needs to be bigger
- [x] Modal needs to be bigger
- [x] Cancel button shouldn't be green

### Implement Send Function

- [x] Send the data for the email (Name, Their Email, Message Text) to a Google Cloud Function 
    _PR https://github.com/holistic-web/portfolio-fabricator/pull/16_
- [x] Write a Google Cloud Function
    _PR https://github.com/holistic-web/portfolio-fabricator/pull/18_

## Future

* Store messages in database
* Allow users to send emails to different addresses
* Allow user to disable email modal

### CODE REVIEW

**Comment by andrewlew1s** [comment]
I forgot to make it a draft PR :(

File: fabricated-website/src/components/AppHeader.vue [hunk 1]
```diff
@@ -9,14 +9,7 @@
...
 			</span>
-
-			<b-button
-				v-if="_get(portfolio, 'contact.email')"
-				class="Header__contactButton"
-				variant="outline-primary"
-				size="lg"
-				:href="`mailto:${portfolio.contact.email}`"
-				v-text="'Get in touch!'"/>
+			<email-modal />
```
Comment (michaelfitzhavey) [inline fabricated-website/src/components/AppHeader.vue]: we might want to use this for other forms of contact (such as direct message), do you think `content-modal` is a better name?

File: fabricated-website/src/components/AppHeader.vue [hunk 2]
```diff
@@ -9,14 +9,7 @@
 				{{portfolio.name.first}} {{portfolio.name.last}}&nbsp;
 				<span class="Header__title--light">- {{portfolio.profession}}</span>
 			</span>
-
```
Comment (michaelfitzhavey) [inline fabricated-website/src/components/AppHeader.vue]: I think this button should still be in the header, following the pattern used in https://github.com/holistic-web/website-gmc/blob/development/src/components/AppHeader.vue#L77-L82. 

You can listen to the emit modal event on the header in `App.vue` similar to the pattern here: https://github.com/holistic-web/website-gmc/blob/development/src/containers/Landing/Page.vue#L3

If we do it that way and keep the modal in the main App page as we have here:
https://github.com/holistic-web/website-gmc/blob/development/src/App.vue#L7-L9

we'll be able to open and close this modal from any component in the site - which could be very useful

File: fabricated-website/src/components/EmailModal.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+				v-text="'Get in touch!'"
+				/>
+
+			<b-modal id="modal-1" title="EmailModal">
+				<p class="my-4">Send mail below!</p>
+				<b-button
+					variant="warning"
+					size="lg"
+					v-text="'CLICK ME'"
+						/>
```
Comment (michaelfitzhavey) [inline fabricated-website/src/components/EmailModal.vue]: formatting looks off here, shouldn't this be on the line above?

File: fabricated-website/src/components/EmailModal.vue [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<template>
+	<div class="EmailModal">
+		<b-button
```
Comment (michaelfitzhavey) [inline fabricated-website/src/components/EmailModal.vue]: As per my comment above, this button isn't part of the contact modal and so shouldn't be in this component. Also the Indentation looks off a little 😄

File: fabricated-website/src/components/EmailModal.vue [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+					variant="warning"
+					size="lg"
+					v-text="'CLICK ME'"
+						/>
+			</b-modal>
+	</div>
+</template>
+
+
+<style lang="scss">
```
Comment (michaelfitzhavey) [inline fabricated-website/src/components/EmailModal.vue]: We don't need empty stylesheets or unused classnames

File: fabricated-website/src/components/AppHeader.vue (hunk 1 above)
Comment (Charlielancaster) [reply fabricated-website/src/components/AppHeader.vue]: what about `contact-modal`? content implies that more than contacting will happen in these modals

Comment (michaelfitzhavey) [reply fabricated-website/src/components/AppHeader.vue]: That's actually what I meant to type and that was a silly typo - thanks @Charlielancaster

Comment (andrewlew1s) [reply fabricated-website/src/components/AppHeader.vue]: fixed in: 1f4ab65

File: fabricated-website/src/components/AppHeader.vue (hunk 2 above)
Comment (andrewlew1s) [reply fabricated-website/src/components/AppHeader.vue]: fixed in: aeaa9ab

Comment (michaelfitzhavey) [reply fabricated-website/src/components/AppHeader.vue]: that's an awesome commit

**Comment by andrewlew1s** [comment]
Updated as per helpful comments

File: fabricated-website/src/components/EmailModal.vue (hunk 3 above)
Comment (andrewlew1s) [reply fabricated-website/src/components/EmailModal.vue]: fixed in: aeaa9ab

File: fabricated-website/src/components/EmailModal.vue (hunk 4 above)
Comment (andrewlew1s) [reply fabricated-website/src/components/EmailModal.vue]: fixed in: aeaa9ab

File: fabricated-website/src/components/EmailModal.vue (hunk 5 above)
Comment (andrewlew1s) [reply fabricated-website/src/components/EmailModal.vue]: fixed in: aeaa9ab

**Comment by andrewlew1s** [comment]
TODO:

- [ ] add form group for user email and message
- [ ] grab email address of recipient

**Comment by michaelfitzhavey** [comment]
@andrewlew1s just fixed some merge conflicts with master, it's all looking really good!

**Comment by michaelfitzhavey** [comment]
#16 is in progress ensuring the UI calls the google firebase function

**Comment by michaelfitzhavey** [comment]
#18 is in progress adding the firebase function to send the email

**Comment by andrewlew1s** [comment]
hooray!

**Pull request merged**